src/ORM/Table.php

```diff
@@ -700,14 +700,27 @@ public function setDisplayField($field)
     public function getDisplayField()
     {
         if ($this->_displayField === null) {
+            $displayFieldFound = false;
             $schema = $this->getSchema();
             $this->_displayField = $this->getPrimaryKey();
-            foreach (['title', 'name', 'label'] as $field) {
+            $expectedFields = ['title', 'name', 'label'];
+            foreach ($expectedFields as $field) {
                 if ($schema->hasColumn($field)) {
                     $this->_displayField = $field;
+                    $displayFieldFound = true;
                     break;
                 }
             }
+            if (!$displayFieldFound) {
+                foreach ($expectedFields as $field) {
+                    foreach ($schema->columns() as $column) {
+                        if (substr($column, -strlen('_' . $field)) === '_' . $field) {
```

Comment:
Should you prepare the field name before looping the columns? You'll do two extra allocations on each loop right now.